server/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -107,8 +107,6 @@ public AlterTableClusterStateExecutor(MetadataMappingService metadataMappingServ
     protected ClusterState execute(ClusterState currentState, AlterTableRequest request) throws Exception {
         if (request.isPartitioned()) {
             if (request.partitionIndexName() != null) {
-                assert request.mappingDelta() == null
```

Comment:
this is not used for add column anymore and explicitly set to null in request in >=5.3